internal/sqlparser/parser.go

```diff
@@ -23,15 +37,37 @@ const (
 	gooseStatementEndDown                      // 6
 )
 
-type stateMachine parserState
+type stateMachine struct {
```

Comment:
I think it makes sense to attach the `verbose bool` to the struct and hang a print helper.

We pass around the `stateMachine` struct as a pointer.